pr_agent/servers/bitbucket_app.py

```diff
@@ -83,16 +83,79 @@ def _get_username(data):
+        if not commits_api:
+            return False
+        if not pull_request.get('updated_on'):
+            return False
+        bearer_token = context.get('bitbucket_bearer_token')
+        headers = {
+            'Authorization': f'Bearer {bearer_token}',
+            'Accept': 'application/json'
+        }
+        response = requests.get(commits_api, headers=headers)
+        if response.status_code != 200:
+            get_logger().warning(f"Bitbucket commits API returned {response.status_code} for {commits_api}")
+            return False
+
+        username =_get_username(data)
+        commits_data = response.json() or {}
+        values = commits_data.get('values') or []
+        if (not values or not isinstance(values, list) or not values[0].get('author') or not values[0]['author'].get('user')
+                or not values[0]['author']['user'].get('display_name')):
+            get_logger().warning("No commits returned for pull request or one of the required fields missing; skipping push validation",
+                                 artifact={'values': values})
+            return False
+        commit_username = commits_data['values'][0]['author']['user']['display_name']
+        if username != commit_username:
+            get_logger().warning(f"Mismatch in username {username} vs. commit_username {commit_username}")
+            return False
+
+        time_pr_updated = pull_request['updated_on']
+        time_last_commit = commits_data['values'][0]['date']
+        from datetime import datetime
+        ts1 = datetime.fromisoformat(time_pr_updated)
+        ts2 = datetime.fromisoformat(time_last_commit)
+        diff = (ts1 - ts2).total_seconds()
+        max_delta_seconds = 15
+        if diff > 0 and diff < max_delta_seconds:
+            is_valid_push = True
+        else:
+            get_logger().debug(f"Too much time passed since last commit",
+                               artifact={'updated': time_pr_updated, 'last_commit': time_last_commit})
+    except Exception as e:
+        get_logger().exception(f"Failed to validate time difference between last commit and PR update",
+                               artifact={'error': e, 'data': data})
+    return is_valid_push
+
 async def _perform_commands_bitbucket(commands_conf: str, agent: PRAgent, api_url: str, log_context: dict, data: dict):
     apply_repo_settings(api_url)
     if commands_conf == "pr_commands" and get_settings().config.disable_auto_feedback:  # auto commands for PR, and auto feedback is disabled
         get_logger().info(f"Auto feedback is disabled, skipping auto commands for PR {api_url=}")
         return
+    if commands_conf == "push_commands":
+        if not get_settings().get("bitbucket_app.handle_push_trigger"):
+            get_logger().info(
+                "Bitbucket push trigger handling disabled via 'bitbucket_app.handle_push_trigger'; skipping push commands")
```

Comment:
/implement
Remove `bitbucket_app.handle_push_trigger'` from the comment to make it easier to maintain in the future. Draft 3 alternatives.